File: apps/mobile/modules/omarchy-link/src/OmarchyLink.types.ts
export type PinnedRequest = {
  url: string;
  method: "GET" | "POST" | "DELETE";
  fingerprint: string;
  headers?: Record<string, string>;
  body?: string;
  timeoutMs?: number;
};

export type PinnedUpload = Omit<PinnedRequest, "body"> & {
  fileUri: string;
};

export type NativeHttpResponse = {
  status: number;
  headers: Record<string, string>;
  body: string;
};

export type DiscoveredDesktop = {
  id: string;
  name: string;
  host: string;
  port: number;
  protocolVersion: number;
};

File: apps/mobile/modules/omarchy-link/ios/OmarchyLinkTypes.swift
import Foundation

struct RequestOptions: Decodable {
  let url: String
  let method: String
  let fingerprint: String
  let headers: [String: String]?
  let body: String?
  let timeoutMs: Double?
  let fileUri: String?
  let uploadId: String?
}

struct TransportResponse: Encodable {
  let status: Int
  let headers: [String: String]
  let body: String
}

struct DiscoveredDesktop: Encodable {
  let id: String
  let name: String
  let host: String
  let port: Int
  let protocolVersion: Int
}

enum OmarchyLinkError: Error, LocalizedError {
  case invalidUrl
  case invalidFingerprint
  case missingResponse
  case responseTooLarge
  case unsupportedFileUrl
  case invalidUploadId

  var errorDescription: String? {
    switch self {
    case .invalidUrl: "The Omarchy desktop address is invalid."
    case .invalidFingerprint: "The desktop certificate fingerprint is invalid."
    case .missingResponse: "The Omarchy desktop returned no HTTP response."
    case .responseTooLarge: "The Omarchy desktop response exceeded 256 KiB."
    case .unsupportedFileUrl: "The selected file cannot be opened by Omarchy Mobile."
    case .invalidUploadId: "The file transfer identifier is invalid."
    }
  }
}
